### PR TITLE
Layout: Expose Provider

### PR DESCRIPTION
**Pull request by andy-hook**
Resolves https://github.com/aragon/aragon-ui/issues/804

### CODE REVIEW

File: src/components/Layout/Layout.js [hunk 1]
```diff
@@ -29,6 +36,10 @@ function getLayoutSize(parentWidth, breakpoints) {
   return sizes[0]
 }
 
+function useMergeBreakpoints(breakpoints) {
+  return useMemo(() => ({ ...BREAKPOINTS, ...breakpoints }), [breakpoints])
```
Comment (bpierre) [inline src/components/Layout/Layout.js:40]: `breakpoints` will probably change on every render if it’s passed to `LayoutProvider`. Should we use `JSON.stringify(breakpoints)` as a key?

( `JSON.stringify()` is probably faster than `Object.keys(a).join() + Object.values(a).join()`. )

File: src/components/Layout/Layout.js [hunk 2]
```diff
@@ -50,12 +61,39 @@ function Layout({
...
+  parentWidth: PropTypes.number,
+}
+
+Layout.defaultProps = {
+  breakpoints: {},
+  paddingBottom: 3 * GU,
+}
+
+// Can be used to build an alternative Layout component
+Layout.__Context = LayoutContext
```
Comment (bpierre) [inline src/components/Layout/Layout.js:92]: If we plan to use it, should we make it part of the public API by removing the `__`?

**Review by bpierre — APPROVED** [review]
LGTM!

File: src/components/Layout/Layout.js (hunk 1 above)
Comment (andy-hook) [reply src/components/Layout/Layout.js:40]: Definitely, good spot!

File: src/components/Layout/Layout.js (hunk 2 above)
Comment (andy-hook) [reply src/components/Layout/Layout.js:92]: I was curious about why this exists. What was the original use case? it feels redundant.

Comment (bpierre) [reply src/components/Layout/Layout.js:98]: Oh I completely forgot I added this, I thought it was a new change. It was added [in this PR](https://github.com/aragon/aragon-ui/pull/538) that was made [for Apiary](https://github.com/1Hive/apiary/pull/52), to do [the exact same thing](https://github.com/1Hive/apiary/blob/3fb0075f32ad418d3f6bc42289c99a98c7de3484/website/src/components/Layout/Layout.js) we need to do now :smile:

Apiary is still using aragonUI 1.0 and this was a private feature, so I think we can just remove it and recommend them to move to `LayoutProvider` whenever they decide to upgrade aragonUI.

**Pull request merged**